jc/parsers/dir.py

```diff
@@ -0,0 +1,332 @@
+        "dir": null,
+        "size": "421,888",
+        "filename": "IEShims.dll"
+      },
+      {
+        "date": "12/06/2019",
+        "time": "02:47 PM",
+        "dir": null,
+        "size": "819,136",
+        "filename": "iexplore.exe"
+      },
+      {
+        "date": "12/07/2019",
+        "time": "02:14 AM",
+        "dir": "<DIR>",
+        "size": null,
+        "filename": "images"
+      },
+      {
+        "date": "12/07/2019",
+        "time": "02:14 AM",
+        "dir": "<DIR>",
+        "size": null,
+        "filename": "SIGNUP"
+      },
+      {
+        "date": "12/07/2019",
+        "time": "02:09 AM",
+        "dir": null,
+        "size": "48,536",
+        "filename": "sqmapi.dll"
+      }
+    ]
+
+"""
+import re
+import jc.utils
+
+
+class info():
+    version = '1.0'
+    description = 'dir command parser'
+    author = 'Rasheed Elsaleh'
+    author_email = 'rasheed@rebelliondefense.com'
+
+    # compatible options: win32
+    compatible = ['win32']
+    magic_commands = ['dir']
+
+
+__version__ = info.version
+
+
+def process(proc_data):
+    """
+    Final processing to conform to the schema.
+
+    Parameters:
+
+        proc_data:   (List of Dictionaries) raw structured data to process
+
+    Returns:
+
+        List of Dictionaries. Structured data with the following schema:
+
+        [
+          {
+            "date": string,
+            "time": string,
+            "dir": string,
+            "size": integer,
+            "filename: string
+          }
+        ]
+    """
+
+    for entry in proc_data:
+        int_list = ["size"]
+        for key in int_list:
+            if entry.get(key):
+                try:
+                    key_int = int(entry[key].replace(",", ""))
+                except ValueError:
+                    entry[key] = None
+                else:
+                    entry[key] = key_int
+    return proc_data
+
+
+def parse(data, raw=False, quiet=False):
+    """
+    Main text parsing function
+
+    Parameters:
+
+        data:        (string)  text data to parse
+        raw:         (boolean) output preprocessed JSON if True
+        quiet:       (boolean) suppress warning messages if True
+
+    Returns:
+
+        List of Dictionaries. Raw or processed structured data.
+
+    """
+    if not quiet:
+        jc.utils.compatibility(__name__, info.compatible)
+
+    raw_output = []
+
+    if jc.utils.has_data(data):
+
+        for line in data.splitlines():
+            # look for first line that starts with a date
+            if not re.match(r'^\d{2}/\d{2}/\d{4}', line):
+                continue
+
+            output_line = {}
+            parsed_line = line.split()
+            output_line["date"] = parsed_line[0]
+            output_line["time"] = " ".join(parsed_line[1:3])
+            output_line.setdefault("dir", None)
+            output_line.setdefault("size", None)
+            if parsed_line[3] == "<DIR>":
+                output_line["dir"] = parsed_line[3]
```

Comment:
I wonder if it makes sense to turn this field into a boolean so if "<DIR>" is detected, then output_line['dir'] = True ? Otherwise, False. This way it's easy to programmatically tell if this is a directory or a file from the one field.

Reply:
yeah that makes more sense. I'll update it.